src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PostMergeBlockProducer.cs

```diff
@@ -77,6 +77,13 @@ protected override Block PrepareBlock(BlockHeader parent, PayloadAttributes? pay
             // TODO: this seems to me that it should be done in the Eth2 seal engine?
             block.Header.ExtraData = Array.Empty<byte>();
             block.Header.IsPostMerge = true;
+
+            if (TrySetState(parent.StateRoot))
+            {
+                Block? processed = ProcessPreparedBlock(block, null);
+                if (processed != null)
+                    block = processed;
+            }
```

Comment:
Why are we processing block here?
That  looks weird to me, this is not the place for that.

Reply:
We need to process here to apply rewards even on empty blocks